api/v1alpha1/backstage_types.go

```diff
@@ -193,6 +201,66 @@ type BackstageList struct {
 	Items           []Backstage `json:"items"`
 }
 
+// Route specifies configuration parameters for OpenShift Route for Backstage.
+// Only a secured edge route is supported for Backstage.
+type Route struct {
+	// Control the creation of a Route on OpenShift.
+	// +optional
+	//+kubebuilder:default=true
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// Host is an alias/DNS that points to the service. Optional.
+	// Ignored if Enabled is false.
+	// If not specified a route name will typically be automatically
+	// chosen.  Must follow DNS952 subdomain conventions.
+	// +optional
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$`
+	Host string `json:"host,omitempty" protobuf:"bytes,1,opt,name=host"`
+
+	// Subdomain is a DNS subdomain that is requested within the ingress controller's
+	// domain (as a subdomain).
+	// Ignored if Enabled is false.
+	// Example: subdomain `frontend` automatically receives the router subdomain
+	// `apps.mycluster.com` to have a full hostname `frontend.apps.mycluster.com`.
+	// +optional
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$`
+	Subdomain string `json:"subdomain,omitempty"`
+
+	// The tls field provides the ability to configure certificates for the route.
+	// Ignored if Enabled is false.
+	// +optional
+	TLS *TLS `json:"tls,omitempty"`
+}
+
+type TLS struct {
+	// certificate provides certificate contents. This should be a single serving certificate, not a certificate
+	// chain. Do not include a CA certificate.
+	Certificate string `json:"certificate,omitempty"`
+
+	// ExternalCertificateSecretName provides certificate contents as a secret reference.
+	// This should be a single serving certificate, not a certificate
+	// chain. Do not include a CA certificate. The secret referenced should
+	// be present in the same namespace as that of the Route.
+	// Forbidden when `certificate` is set.
+	// +optional
+	ExternalCertificateSecretName string `json:"externalCertificateSecretName,omitempty"`
+
+	// key provides key file contents
+	Key string `json:"key,omitempty"`
+
+	// caCertificate provides the cert authority certificate contents
+	CACertificate string `json:"caCertificate,omitempty"`
+}
+
 func init() {
 	SchemeBuilder.Register(&Backstage{}, &BackstageList{})
 }
+
+func (s BackstageSpec) LocalDbEnabled() bool {
+	if s.Database.EnableLocalDb == nil {
```

Comment:
How about a one-line call to `pointer.BoolDeref`?

Reply:
you mean: 
return pointer.BoolDeref(s.Database.EnableLocalDb, true) ?

good option as well.